packages/core/src/directive/recurse.ts

```diff
@@ -0,0 +1,53 @@
+
+import { Param, ParamType } from '../param';
+
+export type RecurseBuilderArgs = {
+  // The loop parameter can be set to false, in which case paths which lead to a loop would be ignored while traversing.
+  loop?: boolean;
+  // the maximum depth to recurse.
+  depth?: number;
+}
+
+export class RecurseBuilder {
```

Comment:
if you are not doing anything in the builder, you can simply not have a builder and use `Recurse` directly in this case without builder.